### PR TITLE
SPA-1879: Extend component definition parsing so it can be used in th…

### PR DESCRIPTION
**Pull request by vitaly-retel**
…e digital-editor

- enable tslint rule eofline
- use package.json "files" property instead of npmignore file

### CODE REVIEW

**Comment by vitaly-retel** [comment]
Component definition model is not correct now:
- component properties,  here can be anonymous property and property object where seems only "name" is required field
I didn't do it because it doesn't look trivial, better to do it as another task.

File: lib/parser/parser-utils.ts [hunk 1]
```diff
@@ -150,3 +210,103 @@ function findComponentPropertyTemplate(
...
+    defaultComponentContent: ComponentSet['defaultComponentContent'],
+    componentName: string,
+    property: ComponentProperty,
+) : void {
+    // No default value
+    if (!property.defaultValue) {
+        return;
+    }
+
+    switch (property.dataType) {
```
Comment (wwsno) [inline lib/parser/parser-utils.ts]: Can you make sure all cases have matching tests? Currently only styles is tested.

File: lib/parser/parser-utils.ts [hunk 2]
```diff
@@ -5,8 +5,55 @@
...
+ * - if group names are unique
+ * - if components have HTML rendition
+ *
+ * @param componentsDefinition
+ * @param getFileContent
+ * @returns Promise<ParsedComponentsDefinition>
+ */
+export async function parseDefinition(
+    componentsDefinition: ComponentsDefinition,
+    getFileContent?: GetFileContentType,
```
Comment (wwsno) [inline lib/parser/parser-utils.ts]: Does it need to be optional? Could we make it required?

Comment (vitaly-retel) [reply lib/parser/parser-utils.ts]: But the componentsDefinition may contain the renditions, then getFileContent is not needed. Or what do you mean? Passing null or empty function?

File: test/parser/parser-util.test.ts [hunk 3]
```diff
@@ -1,226 +1,268 @@
...
+                `<figure doc-image="image"></figure>` +
+                `<div>` +
+                    `<img doc-image="second-image">` +
+                    `<p doc-editable="caption"></p>` +
+                `</div>` +
+            `</div>`;
             getFileContent = (filePath: string) : Promise<string> => {
                 const filename = path.basename(filePath);
                 let result;
                 switch (filename) {
```
[the remark this answers is not in the document]
Comment (vitaly-retel) [reply test/parser/parser-util.test.ts]: ok

File: lib/parser/parser-utils.ts (hunk 2 above)
Comment (wwsno) [reply lib/parser/parser-utils.ts]: Never mind, leave it for now I guess. The design of the components set api in the backend makes it inconsistent.

**Pull request merged**